test/DaemonExec.test.ts

```diff
@@ -0,0 +1,103 @@
+import * as Docker from 'dockerode'
+
+import { promises as fs, createWriteStream, mkdtempSync } from 'fs'
+
+import { resolve, join } from 'path'
+
+import { tmpdir } from 'os'
+import {
+	VolumeDefinition,
+	listImages,
+	listContainers,
+	ensureImageImport,
+	createContainer,
+	attachStreams,
+	stopContainer,
+	removeContainer,
+} from '../src/Daemon/DaemonExec'
+
+interface ExpectedStatus {
+	Error: unknown
+	StatusCode: number
+}
+
+const docker = new Docker()
+jest.setTimeout(300000)
+
+describe('DaemonExec', () => {
+	it('pulls a specified image', async () => {
+		await ensureImageImport(docker, 'ubuntu:18.04')
+		const imageInfos: Docker.ImageInfo[] = await listImages(docker)
+		expect(
+			imageInfos.some((info: Docker.ImageInfo) =>
+				info.RepoTags.some((tag: string) => tag === 'ubuntu:18.04'),
+			),
+		).toBeTruthy()
+	})
+	it('creates a specified container', async () => {
+		await ensureImageImport(docker, 'ubuntu:18.04')
+		const volume: VolumeDefinition[] = [
+			{ fromPath: process.cwd(), toPath: '/test' },
+		]
+		const container: Docker.Container = await createContainer(
+			docker,
+			'ubuntu:18.04',
+			['/bin/ls', '/test'],
+			volume,
+		)
+		expect(container).toBeDefined()
+		expect(container.id).toBeDefined()
+		await removeContainer(container)
+	})
+	it('runs a specified container', async () => {
+		await ensureImageImport(docker, 'ubuntu:18.04')
+		const volume: VolumeDefinition[] = [
+			{ fromPath: process.cwd(), toPath: '/test' },
+		]
+		const container: Docker.Container = await createContainer(
+			docker,
+			'ubuntu:18.04',
+			['/bin/cat', '/test/package.json'],
+			volume,
+		)
+		expect(container).toBeDefined()
+		expect(container.id).toBeDefined()
+
+		const directory = mkdtempSync(join(tmpdir(), 'test'))
+		const target = resolve(directory, 'package.json')
+		const stdoutStream: NodeJS.WritableStream = createWriteStream(target)
+
+		await attachStreams(container, process.stdin, stdoutStream, process.stderr)
```

Comment:
Why attach the Jest process's stdin and stdout streams to this container? Also, testing stdin and stdout in addition to stdout would make it a more thorough test.